docs/gallery/domain/ecephys.py

```diff
@@ -5,152 +5,291 @@
+)
+ecephys_module.add(lfp)
 
 ####################
 # .. _units_electrode:
 #
-# Associate electrodes with units
-# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# Spike Times
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 #
-# The :ref:`PyNWB Basics tutorial <basics>` demonstrates how to add data about units and specifying custom metadata
-# about units. As mentioned :ref:`here <units_fields_ref>`, there are some optional fields for units, one of these
-# is *electrodes*. This field takes a list of indices into the electrode table for the electrodes that the unit
-# corresponds to. For example, if two units were inferred from the first electrode (*id* = 1, index = 0), you would
-# specify that like so:
+# Spike times are stored in the :py:class:`~pynwb.misc.Units` table, which is another subclass of :py:class:`~hdmf.common.table.DynamicTable`.
+# Adding columns to the :py:class:`~pynwb.misc.Units` table is analogous to how we can add columns to the ``"electrodes"`` and ``"trials"`` tables.
+#
+# We will generate some random spike data and populate the :py:class:`~pynwb.misc.Units` table using the :py:class:`~pynwb.file.NWBFile.add_unit`
+# method. Then we can display the :py:class:`~pynwb.misc.Units` table as a pandas :py:class:`~pandas.DataFrame`.
+
+nwbfile.add_unit_column(name='quality', description='sorting quality')
 
-nwbfile.add_unit(id=1, electrodes=[0])
-nwbfile.add_unit(id=2, electrodes=[0])
+poisson_lambda = 20
+firing_rate = 20
+n_units = 10
+for n_units_per_shank in range(n_units):
+    n_spikes = np.random.poisson(lam=poisson_lambda)
+    spike_times = np.round(np.cumsum(np.random.exponential(1 / firing_rate, n_spikes)), 5)
+    nwbfile.add_unit(spike_times=spike_times,
+                     quality='good',
+                     waveform_mean=[1., 2., 3., 4., 5.])
+
+nwbfile.units.to_dataframe()
 
 #######################
 # Designating electrophysiology data
 # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 #
-# As mentioned above, :py:class:`~pynwb.ecephys.ElectricalSeries` and :py:class:`~pynwb.ecephys.SpikeEventSeries`
-# are meant for storing specific types of extracellular recordings. In addition to these two
-# :py:class:`~pynwb.base.TimeSeries` classes, NWB provides some :ref:`data interfaces <basic_data_interfaces>`
+# As mentioned above, :py:class:`~pynwb.ecephys.ElectricalSeries` objects
+# are meant for storing specific types of extracellular recordings. In addition to this
+# :py:class:`~pynwb.base.TimeSeries` class, NWB provides some :ref:`basic_data_interfaces`
 # for designating the type of data you are storing. We will briefly discuss them here, and refer the reader to
-# :py:mod:`API documentation <pynwb.ecephys>` and :ref:`PyNWB Basics tutorial <basics>` for more details on
+# :py:mod:`API documentation <pynwb.ecephys>` and :ref:`basics` for more details on
 # using these objects.
 #
 # For storing spike data, there are two options. Which one you choose depends on what data you have available.
```

Comment:
I was not sure if this section needs to be updated, so I did not remove this _Designating electrophysiology data_ part from the tutorial. Let me know if this should be removed though!